### PR TITLE
[zh-cn]: Update translation of isprototypeof

### PR DESCRIPTION
**Pull request by Irene200825**
### Description
Such as the title.

### Motivation
Ditto.

### Additional details
None.

### Related issues and pull requests
None.

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md [hunk 1]
```diff
@@ -31,35 +33,82 @@ prototypeObj.isPrototypeOf(object)
...
+  }
+  static isFoo(x) {
+    return #value in x;
+  }
+}
+
+const baz = { __proto__: Foo.prototype };
+
+if (Foo.isFoo(baz)) {
+  // 不会运行，因为baz不是Foo
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:110]:
```suggestion
  // 不会运行，因为 baz 不是Foo
```

File: files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md [hunk 2]
```diff
@@ -76,4 +125,4 @@ if (Foo.prototype.isPrototypeOf(baz)) {
 - {{jsxref("Operators/instanceof", "instanceof")}}
 - {{jsxref("Object.getPrototypeOf()")}}
 - {{jsxref("Object.setPrototypeOf()")}}
-- [`Object.prototype.__proto__`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object/proto)
+- [`Inheritance and the prototype chain`](/zh-CN/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:128]:
```suggestion
- [继承与原型链](/zh-CN/docs/Web/JavaScript/Inheritance_and_the_prototype_chain)
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions.

File: files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md [hunk 3]
```diff
@@ -31,35 +33,82 @@ prototypeObj.isPrototypeOf(object)
```
[the remark this answers is not in the document]
Comment (Irene200825) [reply files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:33]: 好的

File: files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md [hunk 4]
```diff
@@ -31,35 +33,82 @@ prototypeObj.isPrototypeOf(object)
...
+
+const baz = { __proto__: Foo.prototype };
+
 if (Foo.prototype.isPrototypeOf(baz)) {
-  // do something safe
+  console.log(Foo.getValue(baz)); // TypeError: Cannot read private member #value from an object whose class did not declare it
+}
+```
+
+对于{{jsxref("Operators/instanceof", "instanceof")}}也是同样的情况。如果您需要以安全的方式读取私有属性，可以使用{{jsxref("Operators/in", "in")}}操作符进行检查。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:94]:
```suggestion
对于 {{jsxref("Operators/instanceof", "instanceof")}} 也是同样的情况。如果你需要以安全的方式读取私有属性，可以提供一个使用 {{jsxref("Operators/in", "in")}} 运算符的品牌检查（branded check）方法。
```

The English document has metioned "branded check", I'm not really sure about the translation, it seems to be a defined [here](https://github.com/tc39/how-we-work/blob/main/terminology.md#brand-check).

@Josh-Cena, Do you have any suggestion about the translation?

Comment (Josh-Cena) [reply files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:94]: The notion of "branded check" comes from type theory ("branded type"). Unfortunately I don't know much about Chinese terminology in advanced computing... Surprisingly, very few posts on the Chinese internet mentioned `#x in obj` at all when I searched for new features in ES2022.

Comment (Irene200825) [reply files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:94]: It is a specialized term. Can it be translated using common terminology?
branded check：对对象的特定标识或属性进行检查？
可以使用 {{jsxref("Operators/in", "in")}} 运算符对对象的特定标识或属性进行检查。

File: files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jasonren0403) [reply files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:33]: “报错”部分需要同步原文

File: files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md (hunk 4 above)
Comment (yin1999) [reply files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md]: As the description in TC39:

> a check against a unique datatype whose creation is controlled by a piece of code.

 I prefer "特定类型检查".

any idea about this @jasonren0403?

Comment (jasonren0403) [reply files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md]: "特定类型检查" seems to be the best among the context. 👍

File: files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md [hunk 5]
```diff
@@ -31,35 +33,82 @@ prototypeObj.isPrototypeOf(object)
...
+
+const baz = { __proto__: Foo.prototype };
+
 if (Foo.prototype.isPrototypeOf(baz)) {
-  // do something safe
+  console.log(Foo.getValue(baz)); // TypeError: Cannot read private member #value from an object whose class did not declare it
+}
+```
+
+对于 {{jsxref("Operators/instanceof", "instanceof")}} 也是同样的情况。如果你需要以安全的方式读取私有属性，可以提供一个使用 {{jsxref("Operators/in", "in")}} 运算符的品牌检查（branded check）方法。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:94]:
```suggestion
对于 {{jsxref("Operators/instanceof", "instanceof")}} 也是同样的情况。如果你需要以安全的方式读取私有属性，可以提供一个使用 {{jsxref("Operators/in", "in")}} 运算符的特定类型检查（branded check）方法。
```

File: files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md [hunk 6]
```diff
@@ -31,35 +33,82 @@ prototypeObj.isPrototypeOf(object)
 
 ## 描述
 
-`isPrototypeOf()` 方法允许你检查一个对象是否存在于另一个对象的原型链上。
+所有继承自 `Object.prototype` 的对象（即除了 [`null` 原型对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object#null_原型对象) 之外的所有对象）都继承了 `isPrototypeOf()` 方法。该方法允许你检查对象是否存在于另一个对象的原型链中。如果作为参数的 `object` 不是对象（即基本类型），则该方法直接返回 `false`。否则，`this` 值被[转换为对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object#对象强制转换)，并且在 `object` 的原型链中搜索 `this` 值，直到到达链的末端或找到 `this` 值为止。
```
Comment (skyclouds2001) [inline files/zh-cn/web/javascript/reference/global_objects/object/isprototypeof/index.md:36]:
```suggestion
所有继承自 `Object.prototype` 的对象（即除了 [`null` 原型对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object#null_原型对象)之外的所有对象）都继承了 `isPrototypeOf()` 方法。该方法允许你检查对象是否存在于另一个对象的原型链中。如果作为参数的 `object` 不是对象（即基本类型），则该方法直接返回 `false`。否则，`this` 值被[转换为对象](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Object#对象强制转换)，并且在 `object` 的原型链中搜索 `this` 值，直到到达链的末端或找到 `this` 值为止。
```

**Review by yin1999 — APPROVED** [review]
Thanks @Irene200825, welcome aboard :tada: ;)

**Pull request merged**